status/private/accounts/accounts.nim

```diff
@@ -80,16 +71,14 @@ proc deleteAccount*(db: DbConn, address: Address) {.raises: [Defect,
                       WHERE       {tblAccounts.address.columnName} = ?"""
 
     db.exec(query, address)
+    ok()
 
-  except SqliteError as e:
-    raise (ref AccountDbError)(parent: e, msg: errorMsg)
-  except ValueError as e:
-    raise (ref AccountDbError)(parent: e, msg: errorMsg)
+  except SqliteError: err OperationError
+  except ValueError: err QueryBuildError
 
-proc getWalletAccount*(db: DbConn, address: Address): Option[Account] {.raises:
-  [AccountDbError, AssertionError, Defect].} =
+proc getWalletAccount*(db: DbConn, address: Address): DbResult[Option[Account]]
```

Comment:
I did also think of this while working on the PR and after some consideration, I feel that it's not the right move because `Result` is a wrapper mainly for return types and errors. It's still valid to successfully return a "none" value, just as it's valid to return an error. For example, if we return a value from the database, the call that retrieves the value from the DB could succeed (therefore no error), yet the value from the DB could be empty, in which case a `none` value would be returned. I think this is a very valid case where replacing it would require an error to be returned, a default value to the provided, or an empty initialised object to be returned, none of which are as accurate as `none` in my opinion.